components/api/db/cellery-hub-db.bal

```diff
@@ -72,4 +73,36 @@ public function getImageVersions(string imageId, int offset, int resultLimit) re
     return res;
 }
 
+public function getPublicArtifact(string orgName, string imageName, string artifactVersion) returns json | error {
+    log:printDebug(io:sprintf("Performing data retrieval for articat \'%s/%s:%s\'", orgName, imageName, artifactVersion));
+    table<record {}> res = check connection->select(GET_ARTIFACT_FROM_IMG_NAME_N_VERSION, gen:ArtifactResponse, orgName, imageName,
+                                                    artifactVersion, loadToMemory = true);
+    return buildJsonPayloadForGetArtifact(res, orgName, imageName, artifactVersion);
+}
+
+public function getUserArtifact(string userId, string orgName, string imageName, string artifactVersion) returns json | error {
+    log:printDebug(io:sprintf("Performing data retrieval for articat \'%s/%s:%s\'", orgName, imageName, artifactVersion));
+    table<record {}> res = check connection->select(GET_ARTIFACT_FOR_USER_FROM_IMG_NAME_N_VERSION, gen:ArtifactResponse, orgName, imageName,
+                                                    artifactVersion, userId, orgName, imageName, artifactVersion, loadToMemory = true);
+    return buildJsonPayloadForGetArtifact(res, orgName, imageName, artifactVersion);
+}
 
+function buildJsonPayloadForGetArtifact(table<record {}> res, string orgName, string imageName, string artifactVersion) returns json | error {
+    if (res.count() == 1) {
+        json resPayload = {};
+        gen:ArtifactResponse artRes = check gen:ArtifactResponse.convert(res.getNext());
+        string metadataString = encoding:byteArrayToString(artRes.metadata);
+        io:StringReader sr = new(metadataString, encoding = "UTF-8");
+        json metadataJson = check sr.readJson();
+        resPayload["description"] = artRes.description;
+        resPayload["pullCount"] = artRes.pullCount;
+        resPayload["lastAuthor"] = artRes.lastAuthor;
+        resPayload["updatedTimestamp"] = artRes.updatedTimestamp;
+        resPayload["metadata"] = metadataJson;
+        return resPayload;      
+    } else {
```

Comment:
This else condition will get hit when res.count is 0 or more than 1.  Is that correct? Or should it be changed to res.count == 0?

Reply:
yes. It should be changed to res.count == 0.